### PR TITLE
fix: public key bundling

### PR DESCRIPTION
**Pull request by erickzanardo**
When bundleling the public key in the shorebird.yaml file on android, we were not including the rest of the file, leaving the yaml incomplete.

This PR adds a fix for that.

### CODE REVIEW

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -1356,6 +1352,19 @@ class FlutterPlugin implements Plugin<Project> {
...
                     }
                   }
+
+                  def shorebirdPublicKeyEnvVar = System.getenv('SHOREBIRD_PUBLIC_KEY')
+                  if (shorebirdPublicKeyEnvVar != null && !shorebirdPublicKeyEnvVar.isEmpty()) {
+                      // When a flavor were used, the content variable will already include
+                      // the app_id and other attributes, since the code above makes sure of that
+                      //
+                      // But when no flavor was used, it will be empty, so we make sure that include
+                      // the original file in the beginning
```
Comment (bryanoltman) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy]: It's possible I'm misunderstanding this, but I'm not sure this is correct. I'm seeing this issue in an app without flavors. The yaml written to the release artifact when providing a public key is:

```yaml
patch_public_key: LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQklqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUEyd2RwRUdidXZsUHNiOWkwcVlyZgpNZWZKbkV3MUJIVGk4U1laVEtyWE92SldtRXBQRTFoV2Zia3ZZelh1NWE5NmdWMXlvY0YzRE13bjA0Vm1SbEtoCkM0QWhzRDBOTDBVTmhZaG90YktHOTFLd2kxdkFYcEhoQ2R6NWdRRUJ3MEsxdUI0Snoreks2V0srMzFQcnlZcHcKTHdieVhOcVhvWThJQUFVUTRTVHNIWVY1dytCTVNpOHBlcFdNUmQ3RFI5UkhjYk5PWmxKdmRCUTVOeHZCNEpONApkUk1xOGNDNzNlejFQOWQ3RGZ3djNUV1kraGU5RW11WExUMlVpdlpTbEhJckdCYTdNRmZxeVVlMnJvMEY3VGUvCkIwc2kxMml0QmJXSXF5Y3ZxY1hqZU9QTm42V0VwcU43SVdqYjlMVWgxNjJKeVlhejVMYi9WZWVKWDhMS3RFbGMKY3dJREFRQUIKLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0tCg==
```

Comment (bryanoltman) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy]: Ahhh I see what the issue is. We currently copy the shorebird.yaml file (without alterations) to the app bundle when flavors aren't in use. The check for `!content.isEmpty()` below only triggers if we've added something. We may want to make that clearer (e.g., always write the shorebird.yaml, updating the app_id/removing flavors as necessary)

Comment (erickzanardo) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy]: Yeah, exactly! I guess I was so focused in the edge cases, that I forgot to handle common case.

Maybe my wording in this comment is confusing? Lmk if you think this comment is causing more harm than improvements

File: packages/shorebird_tests/test/android_test.dart [hunk 2]
```diff
@@ -46,29 +56,34 @@ void main() {
         },
       );
 
-      testWithShorebirdProject(
-        'correctly changes the app id and adds the public key when passed through the environment variable',
-        (projectDirectory) async {
-          const base64PublicKey = 'public_123';
-          projectDirectory.addAndroidFlavors();
-          projectDirectory.addShorebirdFlavors();
+      group('when passed through the environment variable', () {
```
Comment (bryanoltman) [inline packages/shorebird_tests/test/android_test.dart]:
```suggestion
      group('when public key passed through environment variable', () {
```

File: packages/shorebird_tests/test/shorebird_tests.dart [hunk 3]
```diff
@@ -184,14 +188,14 @@ $flavors
...
-  Future<String> getGeneratedShorebirdYaml({String? flavor}) async {
+  Future<YamlMap> getGeneratedShorebirdYaml({String? flavor}) async {
     final decodedBytes =
         ZipDecoder().decodeBytes(apkFile(flavor: flavor).readAsBytesSync());
 
     await extractArchiveToDisk(
         decodedBytes, path.join(this.path, 'apk-extracted'));
 
-    return File(
+    final raw = File(
```
Comment (bryanoltman) [inline packages/shorebird_tests/test/shorebird_tests.dart]:
```suggestion
    final yamlString = File(
```

File: packages/shorebird_tests/test/shorebird_tests.dart [hunk 4]
```diff
@@ -200,5 +204,6 @@ $flavors
         'shorebird.yaml',
       ),
     ).readAsStringSync();
+    return loadYaml(raw) as YamlMap;
```
Comment (bryanoltman) [inline packages/shorebird_tests/test/shorebird_tests.dart]:
```suggestion
    return loadYaml(yamlString) as YamlMap;
```

**Review by bryanoltman — APPROVED** [review]
Left a few minor comments, but LGTM!
